source/docs/software/basic-programming/java-units.rst

```diff
@@ -1,58 +1,122 @@
 The Java Units Library
 ======================
 
-The units library is intended to reduce the number of unit-related bugs in robot programs by making all unit conversions explicitly required when units are used. Users can define measurements using values directly from CAD software or physical measuring tools without needing to worry about storing everything in terms of the same unit.
+The units library is a tool that helps programmers avoid mistakes related to units of measurement. It does this by keeping track of the units of measurement, and by ensuring that all operations are performed with the correct units. This can help to prevent errors that can lead to incorrect results, such as adding a distance in inches to a distance in meters.
 
-.. code-block:: java
+An added benefit is readability and maintainability, which also reduces bugs. By making the units of measurement explicit in your code, it becomes easier to read and understand what your code is doing. This can also help to make your code more maintainable, as it is easier to identify and fix errors related to units of measurement.
+
+The units library has a number of features:
 
-   Measure<Mass> kArmMass = Kilograms.of(1.423); // value taken from CAD
-   Measure<Distance> kArmLength = Inches.of(32.25); // value taken from physical measurement
-   Measure<Angle> kMinArmAngle = Degrees.of(5);
-   Measure<Angle> kArmMaxTravel = Rotations.of(0.45);
+- A set of predefined units, such as meters, degrees, and seconds.
+- The ability to convert between different units.
+- Support for performing arithmetic and comparisons on quantities with units.
+- Support for displaying quantities with units in a human-readable format.
 
 Using the Units Library
 -----------------------
 
 The Java units library is available in the ``edu.wpi.first.units`` package. The most relevant classes are `edu.wpi.first.units.Units <https://github.wpilib.org/allwpilib/docs/beta/java/edu/wpi/first/units/Units.html>`__, which contains a set of predefined units; and `edu.wpi.first.units.Measure <https://github.wpilib.org/allwpilib/docs/beta/java/edu/wpi/first/units/Measure.html>`__, which is used to tag a value with a unit. It is recommended to static import ``edu.wpi.first.units.Units.*`` to get full access to all the predefined units.
 
-The library comes with predefined SI and imperial units for distance, angle, time, velocity and acceleration (both linear and angular), mass, voltage, current, power, energy, and temperature.
+The library comes with many predefined units:
+
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| Type                                                   | Units                                                                                                                  |
```

Comment:
I'd rather not us maintain an independent list of units.

Reply:
This is a valid concern, but in the interest of making this document a one-stop-shop, I think it adds a lot of value. If the maintainers feel we should remove it, I will change the wording to encourage users to follow the Javadoc link to the Units class. Honestly, I missed it when I tried to use the library the first time and I really struggled to figure out how to use velocity measures. It's really the reason I created this PR in the first place.
What would the maintainers like?